AP_NavEKF3: move definition of MAX_EKF_CORES

peterbarker opened the pull request:
if you're not including the AP_NavEKF3 header then you don't get this definition and then you won't be able to compile the DAL.

Ryanf55 commented:
How did you produce the error?
On `master`, this compiles:
`./waf configure --board sitl --disable-EKF3 && nice ./waf plane`

peterbarker commented:
> How did you produce the error? On `master`, this compiles: `./waf configure --board sitl --disable-EKF3 && nice ./waf plane`

Moved the include of `AP_NavEKF3/AP_NavEKF3.h` to a different file in `libraries/AP_AHRS` and added conditional compilation guards around that new header file.  See my `pr/active-ekf-type` branch if you're curious.

This can't be replicated in master ATM - but it is clearly a layering problem.

Ryanf55 commented:
> > How did you produce the error? On `master`, this compiles: `./waf configure --board sitl --disable-EKF3 && nice ./waf plane`
> 
> Moved the include of `AP_NavEKF3/AP_NavEKF3.h` to a different file in `libraries/AP_AHRS` and added conditional compilation guards around that new header file. See my `pr/active-ekf-type` branch if you're curious.
> 
> This can't be replicated in master ATM - but it is clearly a layering problem.

Ah I see, thanks! Good change.
